presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -196,49 +198,66 @@ public ParquetHiveRecordCursor(
+                    longs[columnIndex] = integerPartitionKey(columnValue, columnName);
                 }
                 else if (type.equals(SMALLINT)) {
-                    longs[columnIndex] = smallintPartitionKey(partitionKeyValue, columnName);
+                    longs[columnIndex] = smallintPartitionKey(columnValue, columnName);
                 }
                 else if (type.equals(TINYINT)) {
-                    longs[columnIndex] = tinyintPartitionKey(partitionKeyValue, columnName);
+                    longs[columnIndex] = tinyintPartitionKey(columnValue, columnName);
                 }
                 else if (type.equals(BIGINT)) {
-                    longs[columnIndex] = bigintPartitionKey(partitionKeyValue, columnName);
+                    longs[columnIndex] = bigintPartitionKey(columnValue, columnName);
                 }
                 else if (type.equals(DOUBLE)) {
-                    doubles[columnIndex] = doublePartitionKey(partitionKeyValue, columnName);
+                    doubles[columnIndex] = doublePartitionKey(columnValue, columnName);
                 }
                 else if (isVarcharType(type)) {
-                    slices[columnIndex] = varcharPartitionKey(partitionKeyValue, columnName, type);
+                    slices[columnIndex] = varcharPartitionKey(columnValue, columnName, type);
                 }
                 else if (type.equals(TIMESTAMP)) {
-                    longs[columnIndex] = timestampPartitionKey(partitionKey.getValue(), hiveStorageTimeZone, columnName);
+                    longs[columnIndex] = timestampPartitionKey(columnValue, hiveStorageTimeZone, columnName);
                 }
                 else if (type.equals(DATE)) {
-                    longs[columnIndex] = datePartitionKey(partitionKey.getValue(), columnName);
+                    longs[columnIndex] = datePartitionKey(columnValue, columnName);
                 }
                 else if (isShortDecimal(type)) {
-                    longs[columnIndex] = shortDecimalPartitionKey(partitionKey.getValue(), (DecimalType) type, columnName);
+                    longs[columnIndex] = shortDecimalPartitionKey(columnValue, (DecimalType) type, columnName);
                 }
                 else if (isLongDecimal(type)) {
-                    slices[columnIndex] = longDecimalPartitionKey(partitionKey.getValue(), (DecimalType) type, columnName);
+                    slices[columnIndex] = longDecimalPartitionKey(columnValue, (DecimalType) type, columnName);
                 }
                 else {
-                    throw new PrestoException(NOT_SUPPORTED, format("Unsupported column type %s for partition key: %s", type.getDisplayName(), columnName));
+                    if (partitionKey != null) {
+                        throw new PrestoException(NOT_SUPPORTED, format("Unsupported column type %s for partition key: %s", type.getDisplayName(), columnName));
+                    }
+                    else {
```

Comment:
This can be
```java
throw new PrestoException(NOT_SUPPORTED, format("Unsupported column type %s for %s: %s", type.getDisplayName(), columnName, (partitionKey != null) ? "partition key" : "hidden column"));